src/native/external/zlib-ng/CMakeLists.txt

```
@@ -1,8 +1,6 @@
-cmake_minimum_required(VERSION 3.5.1)
+cmake_minimum_required(VERSION 3.5.1...3.31.0)
```

Was this change meant to be in `cmake_minimum_required`? It seems odd to specify a range for the minimum required version.

---

The cmake folks just bumped the minimum version every after a while which breaks these kind of rigid contracts: https://gitlab.kitware.com/cmake/cmake/-/merge_requests/9875 (doesn't seem like they had a good reason to bump the version..).

As noted in PR description, the best way for us is to not use the FetchContent stuff and don't use their configs, as we have done for libunwind etc.

---

The CMake folks bump the minimum version to ensure that they can eventually remove support for old CMake policies.

---

Can we use `3.29.0` as the upper bound here to ensure we don't accidentally get new CMake policy choices we (and the zlib-ng build) aren't expecting?

```suggestion
cmake_minimum_required(VERSION 3.5.1...3.29.0)
```

We also will need to update `zlib-ng-version.txt` with this patch (and preferably upstream it).

---

@jkoritzinsky I see, well that makes sense. I was looking at their MR, it just pointed to previous similar version bump change.

Regarding "using vendor's cmake", we have our own conventions, best practices, version policies and update timelines. I think it's better to not rely on thirdparty configs and list the sources in our own scripts that we need to build for runtime use-cases. e.g. https://github.com/dotnet/runtime/blob/226ee0323375946d91cbac300bf83433539ae36c/src/native/external/llvm-libunwind.cmake#L7-L10